### PR TITLE
Add tracing documentation

### PR DESCRIPTION
**Pull request by marcobebway**
Confirm these statements before you submit your pull request:

- [x] I have read and submitted the required [Contributor Licence Agreements](https://github.com/kyma-project/community/blob/master/CONTRIBUTING.md#agreements-and-licenses).
- [x] This pull request follows the contributing guidelines, recommended Git workflow, and templates.
- [x] I have tested my changes.
- [x] I have updated the relevant documentation.
---

**Description**

Changes proposed in this pull request:

Add tracing documentation to guide in event troubleshooting, errors in lambda and other scenarios.

**Issue link**

### CODE REVIEW

**Comment by abbi-gaurav** [comment]
The images look a bit blurry. Could this be improved?

File: docs/tracing/docs/030-tracing-showcase.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+![](assets/showcase-search-by-tag-error.png)
+
+---
+
+### I want to ensure tracing for the downstream service calls
+
+Ensure tracing for the downstream service calls when you create a lambda or a microservice which is called by another lambda or a microservice triggered by the Events. For example, a lambda called "test-lambda" triggered by the "order.created" Event calls an "http-db-service" microservice to store the order details.
+
+Do the following to ensure that the spans are shown in the trace:
+1.	Pass the trace context headers. Refer to this documentation.
```
Comment (abbi-gaurav) [inline docs/tracing/docs/030-tracing-showcase.md]: The link to documentation.

File: docs/tracing/docs/030-tracing-showcase.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+* Search for error:
+
+  In this case, you must choose the service which has the error and then use the tag.
+
+![](assets/showcase-search-by-tag-error.png)
+
+---
+
+### I want to ensure tracing for the downstream service calls
```
Comment (abbi-gaurav) [inline docs/tracing/docs/030-tracing-showcase.md]: Should we consider keeping the sample and the details for this part in tracing docs as this is specific to tracing and just have a link for the event tracing docs?

File: docs/tracing/docs/030-tracing-showcase.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+Ensure tracing for the downstream service calls when you create a lambda or a microservice which is called by another lambda or a microservice triggered by the Events. For example, a lambda called "test-lambda" triggered by the "order.created" Event calls an "http-db-service" microservice to store the order details.
+
+Do the following to ensure that the spans are shown in the trace:
+1.	Pass the trace context headers. Refer to this documentation.
+2.	Add the label to "app:{name of lambda or service}" when creating the lambda or service.
```
Comment (abbi-gaurav) [inline docs/tracing/docs/030-tracing-showcase.md]: Could we have code formatting here for the `app:***`?

File: docs/tracing/docs/030-tracing-showcase.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+3.	Name the port "http" or with the prefix "http-" (reference: https://istio.io/help/faq/traffic-management/#naming-port-convention).
```
Comment (abbi-gaurav) [inline docs/tracing/docs/030-tracing-showcase.md]: Do we need a label for both service and deployment?
Is it possible to somehow highlight the relevant parts of the code sample?

File: docs/tracing/docs/030-tracing-showcase.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
+---
+title: Troubleshooting
+type: Troubleshooting
+---
+
+## Tracing Showcase
+
+### I have no micro-service or lambda configured to receive an event
```
Comment (derberg) [inline docs/tracing/docs/030-tracing-showcase.md]: This should be a separate document, but not for tracing. What is the purpose of this document anyway. Why would I like to go to jeager to check event that is not consumed by anything. this is not explained. What is publish-service anyway, some random name or? Without this background info it is hard to classify what type of document it is, for sure not a troubleshooting one

File: docs/tracing/docs/030-tracing-showcase.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+In the trace details, you can see the tags for the "publish-service"
+
+![](assets/showcase-only-publish-detail.png)
+
+---
+
+### I have a micro-service / lambda Event trigger configured
+
+In such case, you will also see the "webhook-service" in the trace along with the "publish-service". There will be more spans from Envoy. You can also see the spans from the lambda or service for a downstream call if the tracing context headers are propagated while making those calls.
```
Comment (derberg) [inline docs/tracing/docs/030-tracing-showcase.md]: same case like in above comment, plus: what is the webhook-service. This is new naming that is not covered by even bus documentation at all. No background info.

Without further review I already suggest this docs are converted into a getting started guide, where you can first describe how to check first events are delivered and then after connecting lambda/service how the situation changes. Please make use of your local technical writer and take a look on current getting started guides and their structure to do the same.

File: docs/tracing/docs/030-tracing-showcase.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+
+Since the Event Bus keeps on retrying to deliver the Event until it is successful, you can see multiple spans for the "webhook-service":
+
+![](assets/showcase-error-detail2.png)
+
+---
+
+### I want to search by tags
+
+It is possible to search the traces using tags. Tags are "key-value" pairs configured for each service.
```
Comment (derberg) [inline docs/tracing/docs/030-tracing-showcase.md]: what should I use? headers here, or ? this is not clear what values I can use there in Tags filtering

File: docs/tracing/docs/030-tracing-showcase.md (hunk 2 above)
Comment (derberg) [inline docs/tracing/docs/030-tracing-showcase.md]: This is not troubleshooting, this should be a separate document of Detail tab that explains what values you need to set on your resources, and use in your application to get full picture from the tooling implemented around the event bus

File: docs/tracing/docs/030-tracing-showcase.md (hunk 5 above)
Comment (abbi-gaurav) [reply docs/tracing/docs/030-tracing-showcase.md]: @derberg I have already left a comment to move it to event-bus documentation. This PR should solely focus on enabling developers to use tracing for issues with event-based triggers or computations.

**Comment by marcobebway** [comment]
@derberg We will check that and close it if that's the case.

File: docs/tracing/docs/030-details-troubleshooting.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+title: Troubleshooting
+type: details
+---
+Using tracing you can troubleshoot different problems you might encounter while using Kyma. Understanding some common scenarios and how the expected traces look like will you have a better grasp on how to quickly use Kyma tracing capabilities to pin point the root cause.
+
+### I have no microservice or lambda configured to receive an event
+
+#### Scenario
+
+There is an Event sent from the external system (for example: The Enterprise Commerce) but there is no lambda or microservice configured with the Event trigger.
```
Comment (abbi-gaurav) [inline docs/tracing/docs/030-details-troubleshooting.md]: Maybe we should mention something else in place of `Enterprise commerce`. @tomekpapiernik is there already a standard term to refer external system?

**Comment by klaudiagrz** [comment]
I guess this PR is supposed to be closed?

**Pull request closed without merging**